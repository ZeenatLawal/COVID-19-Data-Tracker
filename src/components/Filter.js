import PropTypes from 'prop-types';
import { Dropdown } from 'react-bootstrap';

const Filter = ({ handleFilter }) => (
  <Dropdown onSelect={handleFilter}>
    <Dropdown.Toggle className="dropdown">
      Filter
    </Dropdown.Toggle>

    <Dropdown.Menu>
      <Dropdown.Item eventKey="all">All</Dropdown.Item>
      <Dropdown.Item eventKey="99999">&#60;100000</Dropdown.Item>
      <Dropdown.Item eventKey="100000">100000+</Dropdown.Item>
      <Dropdown.Item eventKey="200000">200000+</Dropdown.Item>
      <Dropdown.Item eventKey="300000">300000+</Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
);

Filter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default Filter;
