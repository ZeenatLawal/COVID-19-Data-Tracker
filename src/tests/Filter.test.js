import renderer from 'react-test-renderer';
import Filter from '../components/Filter';

describe('Filter component', () => {
  test('Snapshot test', () => {
    const filter = renderer.create(
      <Filter handleFilter={() => {}} />,
    )
      .toJSON();
    expect(filter).toMatchSnapshot();
  });
});
