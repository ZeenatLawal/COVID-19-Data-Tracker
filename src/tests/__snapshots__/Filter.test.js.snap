// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component Snapshot test 1`] = `
<div
  className="dropdown"
>
  <button
    aria-expanded={false}
    className="dropdown dropdown-toggle btn btn-primary"
    disabled={false}
    id="react-aria9739214247-1"
    onClick={[Function]}
    type="button"
  >
    Filter
  </button>
</div>
`;
